tests/test_publish_draft_services.py

```diff
@@ -1,298 +1,580 @@
+                SupplierFrameworkStub(supplier_id=878040, framework_slug="g-cloud-123", on_framework=True).response(),
+                SupplierFrameworkStub(supplier_id=878400, framework_slug="g-cloud-123", on_framework=True).response(),
+                SupplierFrameworkStub(supplier_id=878401, framework_slug="g-cloud-123", on_framework=True).response(),
             ),
-        ) if x is not None
-    ]
+            "g-cloud-123",
+            with_declarations=None
+        )
+        # 1: submitted, supplier not on framework
+        # 2: not submitted, supplier on framework
+        # 3 & 4: submitted, supplier on framework
+        # 5: submitted, supplier on framework
+        self.draft_service1 = DraftServiceStub(
+            framework_slug="g-cloud-123", id=123, supplierId=878004, status='submitted'
+        ).response()
+        self.draft_service2 = DraftServiceStub(
+            framework_slug="g-cloud-123", id=456, supplierId=878040, status='not-submitted'
+        ).response()
+        self.draft_service3 = DraftServiceStub(
+            framework_slug="g-cloud-123", id=789, supplierId=878400, status='submitted'
+        ).response()
+        self.draft_service4 = DraftServiceStub(
+            framework_slug="g-cloud-123", id=101, supplierId=878400, status='submitted'
+        ).response()
+        self.draft_service5 = DraftServiceStub(
+            framework_slug="g-cloud-123", id=202, supplierId=878401, status='submitted'
+        ).response()
 
+        self.mock_data_api_client.find_draft_services_by_framework_iter.side_effect = [
+            [self.draft_service3, self.draft_service4],
+            [self.draft_service5]
+        ]
 
-@pytest.mark.parametrize("dry_run", (False, True,))
-def test_copy_draft_documents(dry_run):
-    mock_draft_bucket = mock.create_autospec(S3, bucket_name="ducky")
-    mock_draft_bucket.path_exists.return_value = True
+    def test_get_draft_services_fetches_drafts_for_framework_suppliers_only(self):
+        drafts = get_draft_services_iter(self.mock_data_api_client, 'g-cloud-123')
+        assert list(drafts) == [
+            self.draft_service3,
+            self.draft_service4,
+            self.draft_service5,
+        ]
+        assert self.mock_data_api_client.find_draft_services_by_framework_iter.call_args_list == [
+            mock.call('g-cloud-123', status='submitted', supplier_id=878040),
+            mock.call('g-cloud-123', status='submitted', supplier_id=878400),
+            mock.call('g-cloud-123', status='submitted', supplier_id=878401),
+        ]
 
-    mock_documents_bucket = mock.create_autospec(S3, bucket_name="puddeny-pie")
+    def test_get_draft_services_for_draft_ids_file_includes_not_submitted_services(self):
+        self.mock_data_api_client.get_draft_service.side_effect = [
+            {"services": self.draft_service2},
+            {"services": self.draft_service3}
+        ]
```

Comment:
This is raising `StopIteration` because it is being called more times than expected 🤔

Reply:
Aha! Good thing we are testing with Python 3.7 so this was caught 😸

Reply:
So one of the 2 failing was definitely a problematic mock, the other failure was the `draft_ids_file` not being recognised as a file, and splitting it into 6 'lines' rather than 2  🤦‍♀️ Anyway, commit now fixed up.